assets/scripts/passwd-strength-indicator.js

```diff
@@ -0,0 +1,63 @@
+(function (window, document) {
+  const indicator = document.getElementById('password-strength')
+
+  document.getElementById('password').addEventListener('input', function(event) {
+    const strength = getStrength(event.target.value)
+    indicator.value = parseInt(strength.percentage, 10)
+    indicator.setAttribute('class', `pw-${strength.label}`)
+  }, false)
+
+  // Return given password srength as an object {percentage, label}
+  getStrength = function (password) {
+    if (!password && password !== '') {
+      throw new Error('password parameter is missing')
+    }
+    if (!password.length) {
+      return {percentage: 0, label: 'weak'}
+    }
+
+    const charsets = [
+      // upper
+      { regexp: /[A-Z]/g, size: 26 },
+      // lower
+      { regexp: /[a-z]/g, size: 26 },
+      // digit
+      { regexp: /[0-9]/g, size: 10 },
+      // special
+      { regexp: /[!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]/g, size: 30 }
+    ]
+
+    const possibleChars = charsets.reduce(function (possibleChars, charset) {
```

Comment:
Missing a comment here to make things more clear : `// possibleChars measures the char diversity of the password`.

So it might be renamed `charDiversity` ?

Not sure of what I wrote…